### PR TITLE
[Gift card] Add help doc link and change label for translation

### PR DESCRIPTION
**Pull request by ludoboludo**
### PR Summary: 

This is to add a link to the help docs in the editor. So that if the merchant needs more info about how to add a gift card product, it's easy to get to. 
It also changes a label so that if they're editing their language file, the label is a bit clearer. 

### Why are these changes introduced?

Solves a couple issues mentioned in https://github.com/Shopify/dawn/issues/2454

### What approach did you take?

Linked to the help doc explaining how to add a gift card product. 

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |

### Visual impact on existing themes

No visual impact on the theme. Just some added context to the editor. 

### Testing steps/scenarios

- [ ] Make sure everything about gift card works as expected. Go to a product page, edit the buy buttons block and enable the gift card feature. 
- [ ] Go to a gift card product and make sure everything is working. When clicking the checkbox it will show a form. 

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139597217814/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: locales/en.default.schema.json [hunk 1]
```diff
@@ -866,7 +866,7 @@
             },
             "show_gift_card_recipient": {
               "label": "Show recipient form for gift card products",
-              "info": "When enabled, gift card products can optionally be sent to a recipient with a personal message."
+              "info": "When enabled, gift card products can optionally be sent to a recipient with a personal message.[Learn more](https://help.shopify.com/en/manual/products/gift-card-products/add-update-gift-card-products)"
```
Comment (ludoboludo) [inline locales/en.default.schema.json:869]: This is for the featured product section but the setting isn't present on that section. So not sure if we should remove it or leave it there for when it's added in the future 🤔  cc: @antiBaconMachine

File: locales/en.default.json [hunk 2]
```diff
@@ -459,7 +459,7 @@
     "form": {
       "checkbox": "I want to send this as a gift",
       "email_label": "Recipient email",
-      "email_label_optional": "Recipient email (optional)",
+      "email_label_optional_for_no_js_behavior": "Recipient email (optional)",
```
[the remark this answers is not in the document]
Comment (ludoboludo) [reply locales/en.default.json:462]: yeah but that's what is turned into the wording in the admin 😬 
So I wasn't sure if `Email label no js` was clear enough for non technical folks. Not that `Email label optional for no js behaviour` is much clearer but if you're looking for that means on the internet, it seems maybe easier 🤷 

What do you think?

**Review by eugenekasimov — COMMENTED** [review]
Looks good. Let's wait for translations.

**Comment by ludoboludo** [comment]
FYI, I just had to replace the actual link I was using. A new doc was shipped by the gift card team and the link I had before was passing in the language when it shouldn't: 
before
```
https://help.shopify.com/en/manual/products/gift-card-products/add-update-gift-card-products)
```
after
```
https://help.shopify.com/manual/online-store/themes/customizing-themes/add-gift-card-recipient-fields
```

**Pull request merged**